packages/features/bookings/lib/handleNewBooking/loadUsers.ts

```diff
@@ -0,0 +1,102 @@
+import { Prisma } from "@prisma/client";
+import type { IncomingMessage } from "http";
+
+import { orgDomainConfig } from "@calcom/features/ee/organizations/lib/orgDomains";
+import { HttpError } from "@calcom/lib/http-error";
+import logger from "@calcom/lib/logger";
+import { UserRepository } from "@calcom/lib/server/repository/user";
+import prisma, { userSelect } from "@calcom/prisma";
+import { credentialForCalendarServiceSelect } from "@calcom/prisma/selects/credential";
+
+import type { NewBookingEventType } from "../handleNewBooking";
+
+const log = logger.getSubLogger({ prefix: ["[loadUsers]:handleNewBooking "] });
+type Hosts = NewBookingEventType["hosts"];
+
+export const loadUsers = async (
+  eventType: NewBookingEventType,
+  dynamicUserList: string[],
+  req: IncomingMessage
+) => {
+  try {
+    return eventType.id
+      ? await loadUsersByEventType(eventType)
+      : await loadDynamicUsers(dynamicUserList, req);
+  } catch (error) {
+    if (error instanceof HttpError || error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new HttpError({ statusCode: 400, message: error.message });
+    }
+    throw new HttpError({ statusCode: 500, message: "Unable to load users" });
+  }
+};
+
+const loadUsersByEventType = async (
+  eventType: NewBookingEventType
+): Promise<NewBookingEventType["users"]> => {
+  const hosts = eventType.hosts || [];
+  validateHosts(hosts);
+  const users = hosts.map(({ user, isFixed, priority }) => ({
+    ...user,
+    isFixed,
+    priority,
+  }));
+  return users.length ? users : eventType.users;
+};
+
+const validateHosts = (hosts: Hosts): void => {
+  if (!Array.isArray(hosts)) {
+    throw new Error("eventType.hosts is not properly defined.");
+  }
+};
+
+const loadDynamicUsers = async (dynamicUserList: string[], req: IncomingMessage) => {
```

Comment:
I think we could avoid sending the entire `req` object to this function. It needs just `orgSlug`. We can compute orgSlug and pass that directly to it.

**Benefits**
- Makes it easy to unit test as arguments are simpler
- Plus, it is easier to understand and provide it what this function needs.